src/dataframe/dataframe.jl

```diff
@@ -156,6 +156,15 @@ function DataFrame(; kwargs...)
     end
 end
 
+function DataFrame(columns::AbstractVector, cnames::AbstractVector{Symbol};
```

Comment:
Isn't this supposed to be deprecated?

Reply:
We can deprecate it, but it was not deprecated. This works cleanly now:
```
julia> DataFrame(Any[1:3, [1,2,3]], [:a,:b])
3×2 DataFrame
│ Row │ a     │ b     │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 1     │
│ 2   │ 2     │ 2     │
│ 3   │ 3     │ 3     │
```
Why would you deprecate it?

Reply:
By now I mean on currently tagged version of DataFrames.jl

Reply:
Actually there is one call, namely: `DataFrames(Any[[1,2,3], [2,3,4], makeunique=true)` that used to work and will not work now, but I guess no one would use it anyway.

Reply:
Also with this the tests go through cleanly and they failed without it (seems it is good to have tests catching breaking changes 😄).

Reply:
Wasn't it the point of https://github.com/JuliaData/DataFrames.jl/pull/1527? I must be missing something.

Reply:
To my understanding the point of #1527 was to make `DataFrame` constructor in sync with Tables.jl (as without it integration with Tables.jl was broken).

I do not see the benefit of deprecating this constructor (although it is only useful for `AbstractVector{Any}` containing `AbstractVectors`), as Tables.jl passes one positional and zero keyword arguments so it is not conflicting, but if you see it we can add `depwarn` to it (but at least for the time being we should keep it because removing it is breaking).

Reply:
I find it confusing that `Vector{Any}` would be accepted when there are two arguments, but not when there's a single one. Deprecating both methods sounds clearer.

Reply:
Which two methods do you mean to deprecate?

Single `Vector{Any}` is accepted after this PR as well as `Vector{Any}` and column names. Also both were accepted in the past (without deprecation)

Reply:
OK, sorry. I thought this PR only added the old one-argument constructor, but deprecated. Now I see we can perfectly keep it without conflicting with the Tables constructor. Fine then.

Reply:
No problem :) - those things are like "communicating vessels" problem (and I tried to be as non-breaking as reasonable) - that is why I really appreciate a careful review 👍.